client/src/pages/ProductListPage.js

```diff
@@ -1,25 +1,121 @@
-import React from "react";
-import { Container, Row } from "react-bootstrap";
-import { employeeNavbarLinks } from "../Routes";
-import { NavbarComponent } from "../ui-components/NavbarComponent/NavbarComponent";
-
-import "bootstrap/dist/css/bootstrap.min.css";
-
-function ProductListPage(props) {
-  return (
-    <Container>
-      <Row>
-        <NavbarComponent
-          links={employeeNavbarLinks}
-          showShoppingCart
-          shoppingCartItems='1'
-        />
-      </Row>
-      <Row>
-        <h1>ProductListPage</h1>
-      </Row>
-    </Container>
-  );
-}
-
-export { ProductListPage };
+import { React, useEffect, useState } from "react";
+import {
+  Container,
+  Row,
+  Col,
+  Form,
+  CardGroup,
+  FormControl,
+} from "react-bootstrap";
+import { employeeNavbarLinks } from "../Routes";
+import { NavbarComponent } from "../ui-components/NavbarComponent/NavbarComponent";
+import Product from "../services/models/Product";
+import ProductCard from "../ui-components/ProductListpageComponents/ProductCardComponent/ProductCard";
+import { RedButton } from "../ui-components/ProductListpageComponents/RedButtonComponent/RedButton";
+import { RedDropdown } from "../ui-components/ProductListpageComponents/RedDropdownComponent/RedDropdown";
+
+import "../ui-components/ProductListpageComponents/Title.css";
+import "bootstrap/dist/css/bootstrap.min.css";
+
+import { findProducts } from "../services/ApiClient";
+
+function ProductListPage(props) {
+  const [products, setProducts] = useState([]);
+
+  //used for storing the content of the search form
+  const [text, setText] = useState();
+
+  //updated when selected a new category
+  const [category, setCategory] = useState();
+
+  //updated only when clicked on search
+  const [searcString, setSearchString] = useState();
+
+  async function updateProducts(category, searcString) {
+    try {
+      const fromServer = await findProducts(category, searcString);
+      setProducts(fromServer);
+    } catch (err) {
+      console.error(`findProducts() -> couldn't retrieve products: ${err}`);
+    }
+  }

```

Comment:
Call in a different way the input parameters of updateProducts because sonar doesn't like in this way